core/src/main/java/nl/nn/adapterframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -205,11 +207,41 @@ protected JsonObject generateOpenApiJsonSchema(Collection<ApiDispatchConfig> cli
 			}
 		}
 		root.add("paths", paths.build());
-		root.add("components", Json.createObjectBuilder().add("schemas", schemas));
+		JsonObjectBuilder components = Json.createObjectBuilder();
+		components.add("schemas", schemas);
+		root.add("components", components);
 
 		return root.build();
 	}
 
+	private JsonArrayBuilder mapServers(HttpServletRequest request) {
+		JsonArrayBuilder serversArray = Json.createArrayBuilder();
+		String protocol = request.isSecure() ? "https://" : "http://";
+		String suffix = "/api";
+		String env = AppConstants.getInstance().getString("dtap.stage", "LOC");
+		int port = request.getServerPort();
+		// Get load balancer url if exists
+		String loadBalancerUrl = AppConstants.getInstance().getProperty("load.balancer.url", null);
+		if(StringUtils.isNotEmpty(loadBalancerUrl)) {
+			serversArray.add(Json.createObjectBuilder().add("url", loadBalancerUrl + suffix).add("description", "load balancer for " + env + " server"));
+		}
+		// Get details from properties
+		String hostname = AppConstants.getInstance().getString("hostname", null);
```

Comment:
this is a very bad idea... the server request path should be used, or a property, nothing else.